src/components/solution-control/solution-control.tsx

```diff
@@ -74,9 +79,21 @@ export const SolutionControl = ({
         onSelect={setSelected}
         onJoin={onJoin}
         onApply={onApply}
-        onModify={onModify}
+        onModify={solutionName => openModifyDialog({ solutionName })}
         onDelete={solutionName => openDeleteDialog({ solutionName })}
       />
+      <SolutionModifyDialog
+        {...modifyDialog}
+        onClose={closeModifyDialog}
+        onModify={() => {
```

Comment:
move it to a separate callback 'onModify'

Reply:
fixed